src/EMS/PatientFields.js

```diff
@@ -1,15 +1,62 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 import FormCheckbox from '../Components/FormCheckbox';
+import FormComboBox from '../Components/FormComboBox';
 import FormInput from '../Components/FormInput';
 import FormRadio from '../Components/FormRadio';
 import FormRadioFieldSet from '../Components/FormRadioFieldSet';
 import FormTextArea from '../Components/FormTextArea';
 import Heading from '../Components/Heading';
+
 import Ringdown from '../Models/Ringdown';
+import ApiService from '../ApiService';
 
 function PatientFields({ ringdown, onChange }) {
+  const [ambulanceId, setAmbulanceId] = useState([]);
+  const [dispatchCall, setDispatchCall] = useState([]);
+
+  let ambulanceOptionsList = [];
+  let dispatchOptionsList = [];
+
+  useEffect(() => {
+    ApiService.ambulances.getIdentifiers().then((response) => {
+      setAmbulanceId(response.data.ambulanceIdentifiers);
+    });
+    if (ringdown.ambulanceIdentifier) {
+      ApiService.emsCalls.getDispatchCallNumbers(ringdown.ambulanceIdentifier).then((response) => {
+        setDispatchCall(response.data.dispatchCallNumbers);
+      });
+    }
+  }, [ringdown.ambulanceIdentifier]);
+
+  function createOptionsList(listName, listInfo) {
```

Comment:
This function can definitely be refactored- if we're passing in the list itself, and the list objects have a consistent object structure (i.e. they each have an "id", etc), then we shouldn't need to pass in a listName and check on it- notice how the code is almost identical between the two if statements? Doesn't listInfo == ambulanceId and listInfo==dispatchCall depending upon what is being passed in?

Reply:
I've refactored/simplified the code...